project/scratch.py

```diff
@@ -1,16 +1,93 @@
 # print(len([1, 2, 3]) - 1)
 
-a = [ 5, 2, 9, 7, 4, 1, 3, 8, 6, 10 ]
-
-def insertion_sort(arr):
-    for i in range(0, len(arr) - 1):
-        if arr[i] > arr[i+1]:
-            a, b = arr[i], arr[i+1]
-            arr[b], arr[a] = arr[a], arr[b]
-        if i > 0:
-            while i > 0:
-                if arr[i] < arr[i-1]:
-                    print(i)
+# a = [ 5, 2, 9, 7, 4, 1, 3, 8, 6, 10 ]
+a = [item for item in range(30)]
+
+
+# a.sort()
+#binary sory.
+def binary_search(arr, target): 
+    # step 1. we find the midpoint.
+    midpoint = len(arr) // 2
```

Comment:
I didn't know about / vs // myself, great find! 
turns out that
9/4 = 4.5
9//4 = 4